File: CloudFlow/FrontEnd-App/src/app/shared/model/content/document-view-version-bundle.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { DocumentTypeModel } from "./document-type.model";
import { DocumentModel } from "./document.model";
import { WorkflowModel } from "./../bpm/workflow.model";
import { WorkflowStateModel } from "./../bpm/workflow-state.model";

@JsonObject
export class DocumentViewVersionBundleModel {
	
	@JsonProperty("document", DocumentModel)
	public document: DocumentModel = null;

	@JsonProperty("documentType", DocumentTypeModel)
	public documentType: DocumentTypeModel = null;

	@JsonProperty("workflow", WorkflowModel, true)
	public workflow: WorkflowModel = null;

	@JsonProperty("workflowState", WorkflowStateModel, true)
	public workflowState: WorkflowStateModel = null;
}

File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/content/DocumentViewVersionBundleModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.content;

import ro.cloudSoft.cloudDoc.presentation.client.shared.model.bpm.WorkflowModel;
import ro.cloudSoft.cloudDoc.presentation.client.shared.model.bpm.WorkflowStateModel;

public class DocumentViewVersionBundleModel {
	
	private DocumentModel document;
	private DocumentTypeModel documentType;
	private WorkflowModel workflow;
	private WorkflowStateModel workflowState;
	
	public DocumentViewVersionBundleModel() {
	}
	
	public DocumentModel getDocument() {
		return document;
	}
	public void setDocument(DocumentModel document) {
		this.document = document;
	}
	public DocumentTypeModel getDocumentType() {
		return documentType;
	}
	public void setDocumentType(DocumentTypeModel documentType) {
		this.documentType = documentType;
	}
	public WorkflowModel getWorkflow() {
		return workflow;
	}
	public void setWorkflow(WorkflowModel workflow) {
		this.workflow = workflow;
	}
	public WorkflowStateModel getWorkflowState() {
		return workflowState;
	}
	public void setWorkflowState(WorkflowStateModel workflowState) {
		this.workflowState = workflowState;
	}
}
